src/Demo/main.c

```diff
@@ -0,0 +1,182 @@
+/**@file
+ * This file is part of the Leon3 BSP for the Test Environment.
+ *
+ * @copyright 2022-2023 N7 Space Sp. z o.o.
+ *
+ * Test Environment was developed under a programme of,
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// \brief Example program for UART interface
+
+#include "Uart.h"
+#include "UartRegisters.h"
+#include "ByteFifo.h"
+#include "SystemConfig.h"
+#include "Startup_Leon3.h"
+#include <rtems.h>
+
+#define REG_BIT(reg, n) (((reg) >> n) & 1U)
+#define REG_BYTE(reg, n) (((reg) >> (n * 4)) & 0x0FU)
+#define TIMEOUT 4000000
+#define FIFO_SIZE 256
+
+static Uart uart0;
+
+static Uart_Config config;
+
+static int err;
+
+static uint8_t length;
+
+BYTE_FIFO_CREATE(rxByteFifo, FIFO_SIZE);
+
+BYTE_FIFO_CREATE(txByteFifo, FIFO_SIZE);
+
+static inline char
+itoc(uint8_t c)
+{
+    switch(c) {
+        case 10:
+            return 'A';
+        case 11:
+            return 'B';
+        case 12:
+            return 'C';
+        case 13:
+            return 'D';
+        case 14:
+            return 'E';
+        case 15:
+            return 'F';
+        default:
+            return '0' + c;
+    }
+}
+
+static void
+debugAddr(const uint32_t addr)
+{
+    char buff[10] = "0x00000000";
+    for(unsigned char n = 0; n < 8; n++) {
+        uint8_t byte = 0;
+        byte += REG_BYTE(addr, n);
+        buff[9 - n] = itoc(byte);
+    }
+    for(unsigned char n = 0; n < 10; n++) {
+        Uart_write(&uart0, buff[n], TIMEOUT, &err);
+    }
+}
+
+static void
+debugReg(const uint32_t reg)
+{
+    for(char n = 0; n < 32; n++) {
+        uint8_t byte = 0;
+        byte += REG_BIT(reg, n);
```

Comment:
bit?

Reply:
removed (changed method of code testing)